driver/src/contracts/stablex_contract.rs

```diff
@@ -207,82 +212,9 @@ fn encode_execution_for_contract(
 
 #[cfg(test)]
 pub mod tests {
-    use mock_it::Matcher;
-    use mock_it::Matcher::*;
-    use mock_it::Mock;
-
-    use dfusion_core::models::BatchInformation;
-
-    use crate::error::ErrorKind;
-
     use super::*;
     use dfusion_core::models::util::map_from_slice;
-
-    type GetSolutionObjectiveValueArguments = (U256, Matcher<Vec<Order>>, Matcher<Solution>);
-    type SubmitSolutionArguments = (U256, Matcher<Vec<Order>>, Matcher<Solution>, Matcher<U256>);
-
-    #[derive(Clone)]
-    pub struct StableXContractMock {
-        pub get_current_auction_index: Mock<(), Result<U256>>,
-        pub get_auction_data: Mock<U256, Result<(AccountState, Vec<Order>)>>,
-        pub get_solution_objective_value: Mock<GetSolutionObjectiveValueArguments, Result<U256>>,
-        pub submit_solution: Mock<SubmitSolutionArguments, Result<()>>,
-    }
-
-    impl Default for StableXContractMock {
-        fn default() -> StableXContractMock {
-            StableXContractMock {
-                get_current_auction_index: Mock::new(Err(DriverError::new(
-                    "Unexpected call to get_current_auction_index",
-                    ErrorKind::Unknown,
-                ))),
-                get_auction_data: Mock::new(Err(DriverError::new(
-                    "Unexpected call to get_auction_data",
-                    ErrorKind::Unknown,
-                ))),
-                get_solution_objective_value: Mock::new(Err(DriverError::new(
-                    "Unexpected call to get_solution_objective_value",
-                    ErrorKind::Unknown,
-                ))),
-                submit_solution: Mock::new(Err(DriverError::new(
-                    "Unexpected call to submit_solution",
-                    ErrorKind::Unknown,
```

Comment:
I wonder if some of our previously handcrafted error strings and expectations in test will have to be adjusted to whatever comes for free with the automock.

Reply:
I had to add some specific `return_const(Err..)` as not mocking a call defaults to panicking.